src/realm/object-store/sync/sync_session.cpp

```diff
@@ -204,8 +204,10 @@ SyncSession::handle_refresh(const std::shared_ptr<SyncSession>& session)
                 // eg. ClientErrorCode::user_not_found, ClientErrorCode::user_not_logged_in
                 session->handle_bad_auth(session_user, error->error_code, error->message);
             }
-            else if (error->http_status_code &&
-                     (*error->http_status_code == 401 || *error->http_status_code == 403)) {
+            else if (error->http_status_code && (*error->http_status_code == 301 || *error->http_status_code == 401 ||
```

Comment:
don't you also need to test for 302?

Reply:
Reverted this change since the 301 error will not be passed back up through this path. If there is a redirect error, it is passed back as an `app::ClientErrorCode`, which is handled in a different if statement in this function.